README
NAME
    SVN::Statistics

ABSTRACT
    SVN::Statistics - perl module to create subversion statistics

PREREQUISITS
    RRDTool and Log4Perl required

SYNOPSIS
    Sample Script (please also have a look into the samples directory):

       use SVN::Statistics;

DESCRIPTION
    SVN::Statistics - perl module to create subversion statistics.

    I found a product (fisheye http://www.cenqua.com/fisheye/), which has
    most of the features I was looking for; because of this and a lack of
    time to work on this I have decided not to continue with the project. If
    you would like to take over, please let me know and I will give full
    repository access.

EXPORT
    none

SEE ALSO
       http://www.linke.de for my personal homepage and
       http://trac.it-projects.com/svnstats for the project TRAC pages
   
       Please submit bugs at http://bugzilla.it-projects.com
   
       Hosted Subversion Version Control provided by http://svn.it-projects.com
       Checkout the latest version at https://svn.it-projects.com/svn/svnstats

AUTHOR
    Markus Linke, markus.linke@linke.de

COPYRIGHT AND LICENSE
    Copyright 2006 by Markus Linke

    This library is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself.



Changes
------------------------------------------------------------------------
r42 | markus.linke@linke.de | 2006-09-05 14:37:20 -0500 (Tue, 05 Sep 2006) | 1 line


------------------------------------------------------------------------
r41 | markus.linke@linke.de | 2006-09-05 14:36:32 -0500 (Tue, 05 Sep 2006) | 1 line

svn tags were missing
------------------------------------------------------------------------
r39 | markus.linke@linke.de | 2006-09-05 14:29:50 -0500 (Tue, 05 Sep 2006) | 1 line


------------------------------------------------------------------------
r38 | markus.linke@linke.de | 2006-07-15 11:45:41 -0500 (Sat, 15 Jul 2006) | 1 line


------------------------------------------------------------------------


samples/svnstats.pl
#!/usr/local/rt3/perl/bin/perl
$VERSION = sprintf "%d.%05d", '$Revision:   1.4  $' =~ /(\d+)/g;

use strict;
use warnings;
use Murex::Base;
use Data::Dumper::Simple;
use RRDTool::OO;
use IO::Toolkit;
use SVN::Log;

# Initialize Logging -----------------------------------
use Log::Log4perl qw(get_logger);
Log::Log4perl->init_and_watch("log.conf", 60);
my $logger = get_logger();
$logger->info("Program started.");

# Command-line options ---------------------------------
my $repository;
my @extra_options = (
  			{ 
		  		Spec		=>  "repository=s",
		  		Variable  	=> \$repository,
		  		Help		=> "--repository=repository",
		  		Verbose		=> ["--repository=repository",
					    	    "Repository",
				   	   	   ] 
			},
   );

IO::Toolkit::commandline(@extra_options);
$repository="https://svn.it-projects.com/svn/markus" unless defined($repository);

$logger->info("Processing $repository");

my $revs = SVN::Log::retrieve ($repository, 1,'HEAD');
#$logger->debug(Dumper($revs));

foreach my $rev ($revs) {
   $logger->info(Dumper($rev->{revision}->{date}));
}
  
die;
        # Constructor     
    my $rrd = RRDTool::OO->new(
                 file => "myrrdfile.rrd" );

        # Create a round-robin database
    $rrd->create(
         step        => 1,  # one-second intervals
         data_source => { name      => "mydatasource",
                          type      => "GAUGE" },
         archive     => { rows      => 5 });

        # Update RRD with sample values, use current time.
    for(1..5) {
        $rrd->update($_);
        sleep(1);
    }

        # Start fetching values from one day back, 
        # but skip undefined ones first
    $rrd->fetch_start();
    $rrd->fetch_skip_undef();

        # Fetch stored values
    while(my($time, $value) = $rrd->fetch_next()) {
         print "$time: ", 
               defined $value ? $value : "[undef]", "\n";
    }

        # Draw a graph in a PNG image
    $rrd->graph(
      image          => "/var/www/html/mygraph.png",
      vertical_label => 'Commits',
      start          => time() - 10,
      draw           => {
          type   => "area",
          color  => '0000FF',
          legend => "svn.it-projects.com commits over Time",
      }
    );


$logger->info("Program ended.");
1;

__END__

# Below is stub documentation for your module. You'd better edit it!

=head1 NAME

script.pl

=head1 SYNOPSIS

./svnstats.pl

=head1 ABSTRACT

  Abstract of the script
  
=head1 DESCRIPTION

Here comes the full description

=head2 EXPORT

None by default.

=head1 SEE ALSO

   Documentation:   http://linke.de/wiki/index.php/Software_Projects#SVN_STATS:_Subversion_Statistics
   Bug-Reports:     https://bugzilla.it-projects.com
   Source Control:  https://svn.it-projects.com/svn/svnstats

=head1 AUTHOR

Markus Linke, markus.linke@linke.de

=head1 COPYRIGHT AND LICENSE

Copyright 2006 by Markus Linke

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself. 

=cut
    

Statistics.pm
package SVN::Statistics;

# $LastChangedDate: 2006-09-05 14:36:32 -0500 (Tue, 05 Sep 2006) $
# $LastChangedRevision: 41 $ 
# $LastChangedBy: markus.linke@linke.de $ 

use 5.008;
use strict;
use warnings;

use vars qw(@ISA @EXPORT @EXPORT_OK %EXPORT_TAGS $VERSION);
use English;
use Data::Dumper::Simple;
#use RRDTool::OO;
use IO::Toolkit;
use SVN::Log;

require Exporter;

our @ISA = qw(Exporter);
our %EXPORT_TAGS = (
    'all' => [
        qw(

          )
    ]);

our @EXPORT_OK = (@{$EXPORT_TAGS{'all'}});

$VERSION = 0 + sprintf("%3f",((qw$LastChangedRevision: 41 $)[-1])/1000);


sub test {
}

1;

__END__

=head1 NAME

SVN::Statistics

=head1 ABSTRACT

SVN::Statistics - perl module to create subversion statistics

=head1 PREREQUISITS

RRDTool and Log4Perl required

=head1 SYNOPSIS

Sample Script (please also have a look into the samples directory):

   use SVN::Statistics;

=head1 DESCRIPTION

SVN::Statistics - perl module to create subversion statistics. 

I found a product (fisheye http://www.cenqua.com/fisheye/), which has most of
the features I was looking for; because of this and a lack of time to work on
this I have decided not to continue with the project. If you would like to take
over, please let me know and I will give full repository access.

=head1 EXPORT

none

=head1 SEE ALSO

   http://www.linke.de for my personal homepage and
   http://trac.it-projects.com/svnstats for the project TRAC pages
   
   Please submit bugs at http://bugzilla.it-projects.com
   
   Hosted Subversion Version Control provided by http://svn.it-projects.com
   Checkout the latest version at https://svn.it-projects.com/svn/svnstats


=head1 AUTHOR

Markus Linke, markus.linke@linke.de

=head1 COPYRIGHT AND LICENSE

Copyright 2006 by Markus Linke

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself. 

=cut

